### ISSUE TITLE
Failed with gatsby-starter-bootstrap

### ISSUE THREAD

**Issue by gucolin** [angeloocana/gatsby-plugin-i18n#8]
I'm starting a project with https://github.com/jaxx2104/gatsby-starter-bootstrap but wanted to add i18n support using this plugin, but when I followed the instruction by adding 
`resolve: 'gatsby-plugin-i18n',
      options: {
        langKeyForNull: 'any',
        langKeyDefault: languages.defaultLangKey,
        useLangKeyLayout: false
}` to the config it failed to build by throwing the following error at the createPagesStatefully phase: `
error UNHANDLED EXCEPTION

  TypeError: expected a string

  - index.js:12 normalizePath
    [test]/[normalize-path]/index.js:12:11

  - components.js:27 module.exports
    [test]/[gatsby]/dist/redux/reducers/components.js:27:38

  - combineReducers.js:133 combination
    [test]/[redux]/lib/combineReducers.js:133:29

  - createStore.js:178 dispatch
    [test]/[redux]/lib/createStore.js:178:22

  - bindActionCreators.js:7
    [test]/[redux]/lib/bindActionCreators.js:7:12

  - api-runner-node.js:46 doubleBoundActionCreators.(anonymous function)
    [test]/[gatsby]/dist/utils/api-runner-node.js:46:13

  - onCreatePage.js:40 Object.onCreatePage
    [test]/[gatsby-plugin-i18n]/onCreatePage.js:40:3

  - api-runner-node.js:104 runAPI
    [test]/[gatsby]/dist/utils/api-runner-node.js:104:36

  - api-runner-node.js:178
    [test]/[gatsby]/dist/utils/api-runner-node.js:178:33

  - map.js:27
    [test]/[async]/internal/map.js:27:9

  - eachOfLimit.js:64 replenish
    [test]/[async]/internal/eachOfLimit.js:64:17

  - eachOfLimit.js:49 iterateeCallback
    [test]/[async]/internal/eachOfLimit.js:49:17

  - onlyOnce.js:12
    [test]/[async]/internal/onlyOnce.js:12:16

  - map.js:29
    [test]/[async]/internal/map.js:29:13

  - util.js:16 tryCatcher
    [test]/[bluebird]/js/release/util.js:16:23

  - nodeify.js:23 Promise.successAdapter
    [test]/[bluebird]/js/release/nodeify.js:23:30

  - promise.js:566 Promise._settlePromise
    [test]/[bluebird]/js/release/promise.js:566:21

  - promise.js:606 Promise._settlePromiseCtx
    [test]/[bluebird]/js/release/promise.js:606:10

  - async.js:138 Async._drainQueue
    [test]/[bluebird]/js/release/async.js:138:12

  - async.js:143 Async._drainQueues
    [test]/[bluebird]/js/release/async.js:143:10

  - async.js:17 Immediate.Async.drainQueues [as _onImmediate]
    [test]/[bluebird]/js/release/async.js:17:14
`

I've even tried to use your i18n default starter which works in the first time and then after I added a bunch of dependencies, run `npm install` or `yarn` and then `gatsby develop` I got this error as well.

**Comment by angeloocana** [comment]
Hi @gucolin !
Thank you for trying out gatsby-plugin-i18n.

Can I see your project?
I think it should be something related to the content.

I'd love to help.

I found it in your error log:
```
onCreatePage.js:40 Object.onCreatePage
[test]/[gatsby-plugin-i18n]/onCreatePage.js:40:3
```

If you open the file: `./node_modules/gatsby-plugin-i18n/onCreatePage.js`
and go to the line 40, you will see:

```js
deletePage({ path: page.path });
```

Add a console.log before this line and post here the output.
It could be like this:
```js
    console.log('>>>>>>> old page:');
    console.log(page);
    console.log('<<<<<<< old page:');
    console.log('>>>>>>> new page:');
    console.log(newPage);
    console.log('<<<<<<< new page:');

    deletePage({ path: page.path });
```


Let's fix it! No bug will resist! =D

**Comment by gucolin** [comment]
Thanks @angeloocana. You can gastby new gatsby-starter-bootstrap, modify the package.json with  addtional dependencies like this, run yarn or npm and do gatsby develop again, and you could see the error. 

`
"dependencies": {
    "animate.css": "^3.5.2",
    "bootstrap": "^4.0.0-beta.2",
    "gatsby": "^1.9.78",
    "gatsby-image": "^1.0.15",
    "gatsby-link": "^1.6.22",
    "gatsby-plugin-google-analytics": "^1.0.10",
    "gatsby-plugin-i18n": "^0.3.3",
    "gatsby-plugin-i18n-readnext": "^0.0.2",
    "gatsby-plugin-i18n-tags": "^0.0.2",
    "gatsby-plugin-manifest": "^1.0.8",
    "gatsby-plugin-nprogress": "^1.0.7",
    "gatsby-plugin-offline": "^1.0.10",
    "gatsby-plugin-preact": "^1.0.12",
    "gatsby-plugin-react-helmet": "^1.0.6",
    "gatsby-plugin-sass": "^1.0.12",
    "gatsby-plugin-sharp": "^1.6.11",
    "gatsby-plugin-sitemap": "^1.2.5",
    "gatsby-remark-copy-linked-files": "^1.5.13",
    "gatsby-remark-images": "^1.5.22",
    "gatsby-remark-prismjs": "^1.2.8",
    "gatsby-remark-responsive-iframe": "^1.4.12",
    "gatsby-remark-smartypants": "^1.4.8",
    "gatsby-source-filesystem": "^1.5.5",
    "gatsby-transformer-remark": "^1.7.17",
    "gatsby-transformer-sharp": "^1.6.13",
    "intl": "^1.2.5",
    "lodash": "^4.17.4",
    "moment": "^2.19.1",
    "react-intl": "^2.4.0",
    "react-lazyload": "^2.3.0",
    "react-responsive-grid": "^0.3.4",
    "react-tap-event-plugin": "^3.0.2",
    "react-typography": "^0.16.5",
    "wowjs": "^1.1.3"
  },
  "devDependencies": {
    "babel-eslint": "^8.0.1",
    "eslint": "^4.7.0",
    "eslint-config-prettier": "^2.5.0",
    "eslint-plugin-prettier": "^2.2.0",
    "eslint-plugin-react": "^7.3.0",
    "gh-pages": "^1.0.0",
    "lint-staged": "^4.3.0",
    "normalize.css": "^7.0.0",
    "prettier": "^1.7.4",
    "striptags": "^3.1.0",
    "stylelint": "^8.2.0",
    "stylelint-config-standard": "^17.0.0",
    "textlint": "^8.2.1",
    "textlint-rule-preset-ja-spacing": "^2.0.1"
  },
`

The output I got is:

`
⠁ >>>>>>> old page:
{ path: '/',
  component: '/test/src/pages/index.js',
  componentChunkName: 'component---src-pages-index-js',
  layout: 'index',
  jsonName: 'index.json',
  internalComponentName: 'ComponentIndex',
  updatedAt: 1509051892889,
  context: {},
  pluginCreator___NODE: 'Plugin component-page-creator',
  pluginCreatorId: 'Plugin component-page-creator',
  componentPath: '/test/src/pages/index.js' }
<<<<<<< old page:
>>>>>>> new page:
{ path: '/',
  component: '/test/src/pages/index.js',
  componentChunkName: 'component---src-pages-index-js',
  layout: 'index',
  jsonName: 'index.json',
  internalComponentName: 'ComponentIndex',
  updatedAt: 1509051892889,
  context: { slug: '/', langKey: 'en' },
  pluginCreator___NODE: 'Plugin component-page-creator',
  pluginCreatorId: 'Plugin component-page-creator',
  componentPath: '/test/src/pages/index.js' }
<<<<<<< new page:
`

**Comment by angeloocana** [comment]
Fixed!!! Just update to v0.3.4 =D

I don't know why, but after adding your dependencies the error starts to happen.

The problem is here:
```
components.js:27 module.exports
[test]/[gatsby]/dist/redux/reducers/components.js:27:38
```
```js
action.payload.componentPath = normalize(action.payload.component);
```

It is asking for action.payload **.component** wich is equal to page.component.
In order to fix, I just replaced:
```js
deletePage({ path: page.path });
```
With:
```js
deletePage(page);
```

---------------------------------------------------------------

Thank you for opening this issue!
Any problem please let me know.

**Comment by gucolin** [comment]
Excellent. Just tried it and the build works! Gotta figure out how to use it properly now. Very appreciate the fast response.
